0_RootFS/gcc_common.jl

```diff
@@ -357,12 +396,22 @@ function gcc_script(compiler_target::Platform)
             GLIBC_CONFIGURE_OVERRIDES+=( libc_cv_ssp=no libc_cv_ssp_strong=no )
         fi
 
-        if [[ ${COMPILER_TARGET} == riscv64-* ]]; then
-            # Explicitly disable C++
-            # (Disable for all architectures?)
-            GLIBC_CONFIGURE_OVERRIDES+=( CXX=false )
+        # Explicitly disable C++.
+        # If we don't do this, glibc will pick up the host C++
+        # compiler (/usr/bin/g++) to build some C++ files. With this
+        # setting, glibc will build C versions of these files instead.
+        GLIBC_CONFIGURE_OVERRIDES+=( CXX=false )
+
+        # These flags are necessary for GCC 14. GCC 14 defaults to a
+        # modern version of C, too modern for the old glibc libraries we are
+        # trying to build. Various configure tests would fail otherwise. (Why
+        # declare variables or functions if they default to int anyway?)
+        GLIBC_CFLAGS="${CFLAGS} -g -O2"
+        if test -d ${WORKSPACE}/srcdir/gcc-14*; then
```

Comment:
Other memo to self: rather than checking paths on disk, have a variable with the GCC version and check against that.